### PR TITLE
DOC: explain the `mode.chained_assignment` option

### PR DESCRIPTION
**Pull request by sietse**
- [x] closes #xxxx (n/a)
- [x] tests added / passed (yes, Sphinx compiles without errors)
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry (n/a)

Make the explanation of the `mode.chained_assignment` option link to the explanation of chained indexing and assignment. Explain what the three possible settings do.

### CODE REVIEW

File: doc/source/indexing.rst [hunk 1]
```diff
@@ -1833,15 +1833,22 @@ that you've done this:
...
 Evaluation order matters
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
-Furthermore, in chained expressions, the order may determine whether a copy is returned or not.
-If an expression will set values on a copy of a slice, then a ``SettingWithCopy``
-warning will be issued.
-
-You can control the action of a chained assignment via the option ``mode.chained_assignment``,
-which can take the values ``['raise','warn',None]``, where showing a warning is the default.
+When using chained indexing, the order may determine whether a copy is returned
```
Comment (jreback) [inline doc/source/indexing.rst]: say 'order of evaluation'

File: doc/source/indexing.rst [hunk 2]
```diff
@@ -1833,15 +1833,22 @@ that you've done this:
...
+or not.
+
+Pandas has the ``SettingWithCopyWarning`` because assigning to a copy of a
+slice is frequently not intentional, but a mistake caused by chained indexing
+returning a copy where a slice was expected. If you would like pandas to be
```
Comment (jreback) [inline doc/source/indexing.rst]: paragraph break after first sentence

File: doc/source/indexing.rst [hunk 3]
```diff
@@ -1833,15 +1833,22 @@ that you've done this:
...
+more or less trusting about assignment to a chained indexing expression, you
+can set the :ref:`option <options>` ``mode.chained_assignment``, which can take
```
Comment (jreback) [inline doc/source/indexing.rst]: use a bulleted list for these options

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/indexing.rst (hunk 3 above)
Comment (sietse) [reply doc/source/indexing.rst]: Done.

File: doc/source/indexing.rst (hunk 2 above)
Comment (sietse) [reply doc/source/indexing.rst]: Done.

File: doc/source/indexing.rst (hunk 1 above)
Comment (sietse) [reply doc/source/indexing.rst]: I agree that merely 'the order' is not explicit enough. Now that I am looking closer, though, I would say it's not the order of evaluation, but the order in which the user chains the indexing operations (which in turn determines the order the operations are evaluated).

I have turned the sentence into this, which I hope is also in the spirit of your edit:

> When you use chained indexing, the order and type of the indexing operation
partially determine whether the result is a slice into the original object, or
a copy of the slice.

What do you think of that? I don't mean to make this a bikeshedding point, so I will implement whatever you suggest as the final version.

File: doc/source/indexing.rst [hunk 4]
```diff
@@ -1833,15 +1833,27 @@ that you've done this:
...
+
+If you would like pandas to be more or less trusting about assignment to a
+chained indexing expression, you can set the :ref:`option <options>`
+``mode.chained_assignment`` to one of these values:
 
-You can control the action of a chained assignment via the option ``mode.chained_assignment``,
-which can take the values ``['raise','warn',None]``, where showing a warning is the default.
+* ``'raise'`` means pandas will raise a ``SettingWithCopyException`` exception
+  you have to deal with.
+* ``'warn'``, the default, means a ``SettingWithCopyWarning`` is printed.
```
Comment (jreback) [inline doc/source/indexing.rst]: put the default first

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. minor change. ping on green.

**Comment by sietse** [comment]
@jreback I've put the default first, and the checks have passed. Does 'ping on green' mean to send you a message when this is done? If yes, this is your ping.

I forgot to say this last time I replied, so I'll say it now: thank you very much! Not just for the review, but for all the work you do for the pandas project. It is very much appreciated.

**Comment by jorisvandenbossche** [comment]
@sietse Thanks a lot! (I fixed the conflict, due to another PR that was just merged)

**Pull request merged**